include/seqan3/io/detail/out_file_iterator.hpp

```diff
@@ -0,0 +1,180 @@
+ * documentation on output iterators](http://en.cppreference.com/w/cpp/concept/OutputIterator).
+ *
+ * This class template differs from std::back_insert_iterator only in that it performs
+ * no checks itself on the assigned values and that it allows comparisons against
+ * ranges::default_sentinel.
+ */
+template <typename file_type>
+class out_file_iterator
+{
+    static_assert(!std::is_const_v<file_type>,
+                  "You cannot iterate over const files, because the iterator changes the file.");
+public:
+    /*!\name Member types
+     * \brief The associated types are `void`, see the full description.
+     * \{
+     */
+    using value_type        = void;
+    using reference         = void;
+    using const_reference   = void;
+    using size_type         = void;
+    using difference_type   = std::ptrdiff_t;
+    //!\brief Tag this class as an input access iterator.
+    using iterator_category = std::output_iterator_tag;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment.
+     * \{
+     */
+    //!\brief Default constructor.
+    constexpr out_file_iterator() = default;
+    //!\brief Construct by host.
+    constexpr out_file_iterator(file_type & _host) noexcept :
+        host{&_host}
+    {}
+    //!\brief Copy constructor.
+    constexpr out_file_iterator(out_file_iterator const &) = default;
+    //!\brief Copy construction via assignment.
+    constexpr out_file_iterator & operator=(out_file_iterator const &) = default;
+    //!\brief Move constructor.
+    constexpr out_file_iterator (out_file_iterator &&) = default;
+    //!\brief Move assignment.
+    constexpr out_file_iterator & operator=(out_file_iterator &&) = default;
+    //!\brief Use default deconstructor.
+    ~out_file_iterator() = default;
+    //!\}
+
+    /*!\name Iterator operations
+     * \{
+     */
+    //!\brief This is a no-op, returns reference to self.
+    out_file_iterator & operator++()
+    {
+        return *this;
+    }
+
+    //!\brief This is a no-op, returns copy of self. In contrast to input iterators, the return type is required.
+    out_file_iterator operator++(int)
+    {
+        return *this;
+    }
+
+    //!\brief Return reference to self.
+
+    out_file_iterator & operator*() noexcept
+    {
+        return *this;
+    }
+
+    /*!\brief Insert the given value into the file, via the file's `push_back()` member.
+     * \tparam arg_t    The argument type (the file's `push_back()` will enforce certain constraints).
+     * \param[in] arg   The argument to be inserted.
+     */
+    template <typename arg_t>
+    out_file_iterator & operator=(arg_t && arg)
+    {
+        assert(host != nullptr);
+        host->push_back(std::forward<arg_t>(arg));
+        return *this;
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \brief This iterator never equals its sentinel.
+     * \{
+     */
+    constexpr bool operator==(ranges::default_sentinel const &) const noexcept
```

Comment:
Can you elaborate, why you think we need these?
It would be possible to define a for loop over an output file. Do we really want that?

Reply:
You could stream over zipped in- and out-iterators in a for loop. I think that's one of the few cases where I would actually use the iterator. Otherwise we could drop the iterator interface altogether and just recommend people call `push_back()` or `emplace_back()` on the file...

Reply:
♐️